assets/js/modules/adsense/components/common/AdBlockingRecoveryToggle.test.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * AdBlockingRecoveryToggle component tests.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import AdBlockingRecoveryToggle from './AdBlockingRecoveryToggle';
+import {
+	render,
+	provideModules,
+	fireEvent,
+} from '../../../../../../tests/js/test-utils';
+import {
+	AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED,
+	MODULES_ADSENSE,
+} from '../../datastore/constants';
+import { ACCOUNT_STATUS_READY, SITE_STATUS_READY } from '../../util';
+import { enabledFeatures } from '../../../../features';
+
+describe( 'AdBlockingRecoveryToggle', () => {
+	const validSettings = {
+		accountID: 'pub-12345678',
+		clientID: 'ca-pub-12345678',
+		useSnippet: false,
+		accountStatus: ACCOUNT_STATUS_READY,
+		siteStatus: SITE_STATUS_READY,
+		adBlockingRecoverySetupStatus: '',
+	};
+
+	it.each( [
+		[ 'the Ad blocker detection feature flag is not enabled', false ],
+		[ 'Ad blocker recovery setup status is empty', true ],
+	] )(
+		'should not render the Ad blocking recovery toggle when %s',
+		( testName, adBlockerDetectionEnabled ) => {
+			const { container } = render( <AdBlockingRecoveryToggle />, {
+				setupRegistry: ( registry ) => {
+					if ( adBlockerDetectionEnabled ) {
+						enabledFeatures.add( 'adBlockerDetection' );
+					}
```

Comment:
There used to be some cases where this was necessary to do, but that's no longer the case and so features should always be passed through the `features` property of the render options instead.